docs/source/tutorials/tic-tac-toe.rst

```diff
@@ -565,4 +565,246 @@ Now if you compile and run your program you should have the following result:
+        if (index < board_.size() && board_[index] == cell_state::empty) {
+
+            //! Change state of the cell to the current player
+            board_[index] = static_cast<cell_state>(player_turn_);
+
+            //! Create x or o based on the current player
+            player_turn_ == x ? create_x(entity_registry_, row, column) : create_o(entity_registry_, row, column);
+
+            //! Switch player
+            player_turn_ = (player_turn_ == player::x) ? player::o : player::x;
+        }
+    }
+
+We call ``play_turn`` with the position of the mouse click.
+ 
+.. code-block:: cpp
+
+   void on_mouse_button_pressed(const event::mouse_button_pressed &evt) noexcept
+    {
+        if (current_game_state_ == running) {
+            //! Retrieve game constants.
+            auto constants = entity_registry_.ctx<tic_tac_toe_constants>();
+
+            //! Play one turn of the Tic-Tac-Toe
+            play_turn(evt.y / constants.cell_height, evt.x / constants.cell_width);
+        } else {
+            //! Here we reset the game
+        }
+    }
+
+Now we assign the ``on_mouse_button_pressed`` event to the mouse click in the constructor.
+
+.. code-block:: cpp
+
+    tic_tac_toe_logic(entt::registry &registry, entt::entity grid_entity, std::vector<cell_state> board) noexcept
+            : system(registry), grid_entity_(grid_entity), board_(std::move(board))
+    {
+        //! stateless system
+        this->disable();
+
+        //! subscribe to mouse_button event
+        this->dispatcher_.sink<event::mouse_button_pressed>().connect<&tic_tac_toe_logic::on_mouse_button_pressed>(
+                *this);
+    }
+
+In the same class, we have enums ``game_state`` and ``player`` too.
+
+.. code-block:: cpp
+
+    //! Private enums
+    enum game_state
+    {
+        running,
+        player_x_won = 1,
+        player_y_won = 2,
+        tie,
+        quit
+    };
+
+    enum player
+    {
+        x = 1,
+        o = 2
+    };
+
+And the other member variables we used such as grid, state board, game state and player turn.
+
+.. code-block:: cpp
+
+    //! Private members variable
+    entt::entity grid_entity_{entt::null};
+    std::vector<cell_state> board_;
+    game_state current_game_state_{game_state::running};
+    player player_turn_{player::x};
+
+After the class definition, we give a name to our system, out of the class scope.
+
+.. code-block:: cpp
+
+    //! Give a name to our system
+    REFL_AUTO(type(tic_tac_toe_logic));
+
+In the constructor of the game_scene we defined before, we create the board and the logic system.
+
+.. code-block:: cpp
+
+    //! Create the board of the tic tac toe
+    auto board = create_board(tictactoe_constants.nb_cells_per_axis);
+
+    //! Create our logic game system and give the fresh grid entity and the fresh board.
+    this->system_manager_.create_system<tic_tac_toe_logic>(grid_entity, board);
+
+Game logic, board and x o themselves, all of them are now complete.
```

Comment:
Good to add the whole code result and a screenshot of the result with litteralinclude for the code and image for the screenshot